staging/src/k8s.io/client-go/tools/cache/controller_test.go

```diff
@@ -295,7 +299,21 @@ func TestHammerController(t *testing.T) {
 
 	// Let's wait for the controller to finish processing the things we just added.
 	// TODO: look in the queue to see how many items need to be processed.
-	time.Sleep(100 * time.Millisecond)
+	err := wait.PollUntilContextTimeout(ctx, 100*time.Millisecond, wait.ForeverTestTimeout, false, func(ctx context.Context) (bool, error) {
+		outputSetLock.Lock()
+		defer outputSetLock.Unlock()
+
+		// Count total operations across all keys
+		var processedOps uint64
+		for _, ops := range outputSet {
+			processedOps += uint64(len(ops))
+		}
+
+		return processedOps >= atomic.LoadUint64(&totalOps), nil
```

Comment:
If I understand the test correctly I think you don't need this new atomic int all , just check `len(outputSet)`, as the outputset length should be the number of processed Ops

It also should be equal, if is greater than I think we have a bug somewhere

Reply:
hmm, I misunderstood the test sorry, is not the number of operations always equal to 100, per the for loop? 
is not enough to make that 100 a constant and reference it in the for loop and here?